frontend/src/PublishMessageTile.tsx

```diff
@@ -0,0 +1,132 @@
+export class PublishMessageTile extends React.Component<SendTextForm, SendTextForm>{
+    constructor(props: SendTextForm, state: SendTextForm) {
+        super(props);
+        this.state = props;
+
+        this.state.eventBus.register(EventBusEventType.SELECTED_POST_CHANGED, (eventType, eventData?) => this.selectPost(eventData));
+    }
+
+    /** Event handler for when a post had been selected, e.g. via PublishedPostsTile.tsx */
+    private selectPost(eventData?: any) {
+        var platforms = new Set<string>();
+        platforms.add("" + eventData.platformId);
+        this.setState({
+            message: eventData.text,
+            platforms: platforms,
+            eventBus: this.state.eventBus
+        });
+    }
+
+    /** Send the request to the back end triggerin a post on the selected platforms. */
+    private send() {
+        fetch('http://localhost:8080/api/publish', {
+            method: 'POST',
+            headers: new Headers({
+                'Content-Type': 'application/json'
+            }),
+            credentials: 'include',
+            body: JSON.stringify({ message: this.state.message, platforms: Array.from(this.state.platforms.values()) })
+        })
+            .then(response => {
+                response.json();
+            })
+            .then(data => {
+                console.log("Result: " + data);
+                this.state.eventBus.fireEvent(EventBusEventType.REFRESH_POSTS);
+            });
+    }
+
+    private formTextAreaUpdated(event: React.ChangeEvent<HTMLTextAreaElement>) {
+        this.update(event.currentTarget.id, event.currentTarget.value);
+    }
+
+    private formInputFieldUpdated(event: React.ChangeEvent<HTMLInputElement>) {
+        this.update(event.currentTarget.id, event.currentTarget.value);
+    }
+
+    private update(id: string, value: string) {
+        if (id === "textMsg") {
+            this.setState({
+                message: value,
+                platforms: this.state.platforms,
+                eventBus: this.state.eventBus
+            });
+        }
+        if (id === "facebook" || id === "twitter") {
+            this.updateCheckbox(id, value);
+        }
+    }
+
+    private updateCheckbox(id: string, value: string) {
+        var platforms = this.state.platforms;
+        if (!platforms) {
+            platforms = new Set<string>();
+        }
+        if (value) {
+            platforms.add(id);
+        } else {
+            platforms.delete(id);
+        }
+
+        this.setState({
+            message: this.state.message,
+            platforms: platforms,
+            eventBus: this.state.eventBus
+        });
+    }
+
+    private createCheckbox(id: string, checked: boolean) {
+        if (checked) {
+            return <input id={id} type="checkbox" onChange={this.formInputFieldUpdated.bind(this)} checked />
+        }
+        return <input id={id} type="checkbox" onChange={this.formInputFieldUpdated.bind(this)} />
```

Comment:
I guess, this would suffice:
```suggestion
        return <input id={id} type="checkbox" onChange={this.formInputFieldUpdated.bind(this)} checked={checked} />
```

Reply:
done

Reply:
Actually it does not. Feel free to try it out :)

Reply:
Well, there might be a linter error or the like, bit I guess, the approach was right &hellip; I'll check that :)